mesheryctl/internal/cli/root/mesh/validate.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2019 Anirudh Jain
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mesh
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"fortio.org/fortio/log"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+// Operation is the common body type to be passed for Mesh Ops
+type Operation struct {
+	Adapter    string `json:"adapter"`
+	CustomBody string `json:"customBody"`
+	DeleteOp   string `json:"deleteOp"`
+	Namespace  string `json:"namespace"`
+	Query      string `json:"query"`
+}
+
+var spec string
+var adapterURL string
+var namespace string
+var token string
+var mesheryProvider string
+var err error
+
+// validateCmd represents the service mesh validation command
+var validateCmd = &cobra.Command{
+	Use:   "validate",
+	Short: "Validate Service Mesh",
+	Args:  cobra.NoArgs,
+	Long:  `Validate Service Meshes based on specified specification.`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		path := "http://localhost:9081/api/mesh/ops"
+		method := "POST"
+
+		data := url.Values{}
+		data.Set("adapter", adapterURL)
+		data.Set("customBody", "")
+		data.Set("deleteOp", "")
+		data.Set("namespace", namespace)
+
+		// Choose which specification to use for conformance test
+		switch spec {
+		case "smi":
+			{
+				data.Set("query", "smiConformanceTest")
+				break
+			}
+		default:
+			{
+				return errors.New("Specification not found")
+			}
+		}
+
+		payload := strings.NewReader(data.Encode())
+
+		client := &http.Client{}
+		req, err := http.NewRequest(method, path, payload)
+
+		if err != nil {
+			return err
+		}
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded;charset=UTF-8")
+		req.AddCookie(&http.Cookie{
+			Name:  "meshery-provider",
+			Value: mesheryProvider,
+		})
+		req.AddCookie(&http.Cookie{
+			Name:  "token",
+			Value: token,
+		})
+
+		res, err := client.Do(req)
+		if err != nil {
+			return err
+		}
+		defer res.Body.Close()
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			return err
+		}
+
+		log.Infof(string(body))
+		log.Infof("Mesh Validation check started")

```

Comment:
Why is this displayed below the result?

Reply:
Hmm... It was there to prevent it from testing things up
Nvm It'll go up now